Recommend line-awesome add-on for V24 (#2477)
vaadin/docs: articles/configuration/development-mode/index.asciidoc

- Custom theme packaged as a `JAR` dependency is added, if it defines any assets to be added to the project; or
- Exported web component is added.

> Custom theme files are served separately from the front-end bundle. They can be modified on the fly; the browser live-reload will trigger refresh immediately -- no re-compilation is needed.

Reviewer: I would perhaps keep this as it was or change the wording all together as
```suggestion
Custom theme files are served separately from the front-end bundle. They can be modified on the fly; the browser live-reload instantly triggers a refresh of the view -- no re-compilation is needed.
```

When customizing an <<{articles}/styling/application-theme#,Application Theme>> in this mode, the following points should be taken into account:

> - Loading CSS into the shadow DOM of Vaadin components by placing them in the theme's `components` sub-folder is not currently supported.

Reviewer: Should we add here `without compilation/generation/bundling/()what ever it is called)`

Author: Yeah, we need to clarify this. Just "is not supported" sounds like it cannot be used at all, but in fact it can be used with hotdeploy mode enabled or if one forces dev-bundle to be re-build anyhow.

Author: Added a clarification about starting a dev-server in this case. Recommending the rebuild doesn't make sense IMO.

- Vaadin component styles are set up in [filename]`styles.css` or in the imported stylesheet file located in the theme directory, with the `::part()` selector.
- Use the `assets` block in the [filename]`theme.json` file to specify the external assets -- fonts, images, or stylesheets -- to be added to the application development bundle, so that they can be used in the application.
- Use Vaadin https://vaadin.com/directory/component/line-awesome-icons-for-vaadin[Line-Awesome] add-on that doesn't require building a new bundle.

This is the default mode.

== Front-End Development Server for Easier Debugging

> When working with a front-end-heavy project, there can be a front-end development server running for fast compilation and live-reload when the front-end files are changed.

Reviewer: ```suggestion
When working with a frontend-heavy project, there can be a front-end development server running for fast compilation and live-reload when the front-end files are changed.
```
as vale suggests
